cmd/utils.go

```diff
@@ -160,6 +161,24 @@ var globalProfiler interface {
 	Stop()
 }
 
+// Prints a given message after a requested delay.
```

Comment:
This looks like it will work:

```go
// Prints a given message after a requested delay. The printing can be
// (optionally) cancelled by writing to or closing the returned
// channel. If `optFollowUp` is non-empty, then the returned channel
// is used as a signal to print a follow-up message (i.e. when a
// signal is received, if the printing has occurred, the follow-up
// message will be printed.)
func delayPrint1(message string, delay time.Duration, optFollowUp string) chan struct{} {
	cancelCh := make(chan struct{}, 1)
	go func() {
		select {
		case <-cancelCh:
		case <-time.After(delay):
			console.Print(message)
			// Is there a follow-up message to print?
			if len(optFollowUp) > 0 {
				// Listen for the signal on the cancel
				// channel
				<-cancelCh
				// Print follow-up message
				console.Print(msgFollowUp)
			}
		}
		return
	}()
	return cancelCh
}
```

Usage:

```go
cancelCh := delayPrint1("Acquiring lock...", time.Second, "done acquiring lock!")
// Do something that *may* take more than a second.
//...
//...
// Now close the channel.
close(cancelCh)
```

Notice that the function also works as a delayed printer too:

```go
delayPrint1("This line was asked to be printed just a second ago", time.Second, "")
// do something for at least a second for the message to appear.
time.Sleep(1)
```